.github/workflows/rust.yml

```diff
@@ -23,130 +23,102 @@ jobs:
     env:
       RUSTFLAGS: "-D warnings"
     steps:
-      - name: Install Dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y build-essential clang ocl-icd-opencl-dev
       - name: Checkout Sources
         uses: actions/checkout@v3
+      - name: Setup sccache
+        uses: hanabi1224/sccache-action@v1.2.0
+        with:
+          cache-key: ${{ runner.os }}-sccache-test-${{ hashFiles('rust-toolchain.toml') }}
+          cache-update: ${{ github.event_name != 'pull_request' }}
+      - name: Apt Dependencies
+        run: sudo make install-deps
       - name: install nextest
         uses: taiki-e/install-action@nextest
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
-      - name: Cargo Check
-        run: cargo check
-      - name: Make Test-All
-        run: make test-all
-
-  lint:
-    name: Lint
-    runs-on: ubuntu-latest
-    steps:
-      - name: Apt Dependencies
-        run: |
-          sudo apt-get update
-          sudo apt-get install -y build-essential clang ocl-icd-opencl-dev
-      - name: Checkout Sources
-        uses: actions/checkout@v3
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
-      - name: Install taplo (TOML linter)
-        run: cargo install taplo-cli --locked
-      - name: Run Linters
-        run: make lint
-
-  audit:
-    name: Audit
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout Sources
-        uses: actions/checkout@v3
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
-      - name: Install Audit
-        run: cargo install cargo-audit
-      - name: Run Audit
-        run: make audit
-
-  unused_dependencies:
-    name: Unused dependencies
-    runs-on: ubuntu-latest
-    steps:
-      - name: Checkout Sources
-        uses: actions/checkout@v3
-      - name: Rust Cache
-        uses: Swatinem/rust-cache@v2
-        timeout-minutes: ${{ fromJSON(env.CACHE_TIMEOUT_MINUTES) }}
-        continue-on-error: true
-      - name: Install udeps
-        run: cargo install cargo-udeps --locked
-      - name: Run udeps
-        run: make udeps
+      - run: make test-all
+        env:
+          CC: "sccache clang"
```

Comment:
not sure if it wouldn't clash with some other steps, but would adding those envs as global for the entire workflow be fine?

Reply:
`sccache` is not always the first step to avoid generating cache for `cargo install [tools]`, so that `sccache` is not available during `cargo install` and `apt-get install`